test/common/run-tests

```diff
@@ -40,6 +40,15 @@ def test_name(test):
     return "{0} {1} {2}{3}".format(test.test_id, test.command[0], test.command[-1], " [ND]" if test.nondestructive else "")
 
 
+def flush_stdout():
+    while True:
+        try:
+            sys.stdout.flush()
+            break
+        except BlockingIOError:
+            time.sleep(0.1)
```

Comment:
Hm, this hangs forever now, right?

Reply:
Not forever, just until the stdout buffer becomes sufficiently drained  again. It's the same approach as in #14225, just "more of the same".

Reply:
I meant when the other side actually hangs and stops reading. We do have global timeouts for this though, and as you said, we're fixing it the same way in other places.

Reply:
When the receiver dies, it  should get `EPIPE` or `SIGPIPE`, not `EAGAIN`. That part I can actually reproduce locally, and confirm that it behaves sensibly. I just can't for the life of me reproduce a `BlockingIOError` situation, I just rely on throwing this against RH CI and seeing that it works.